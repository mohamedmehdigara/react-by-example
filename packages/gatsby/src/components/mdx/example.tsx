import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
