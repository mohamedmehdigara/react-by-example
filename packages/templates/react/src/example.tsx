import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
