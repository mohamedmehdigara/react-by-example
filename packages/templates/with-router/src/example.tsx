import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
