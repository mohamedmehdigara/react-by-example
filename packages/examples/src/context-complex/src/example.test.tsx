import {
  cleanup,
  fireEvent,
  render,
  RenderResult,
} from '@testing-library/react'
import React, { FC, useContext } from 'react'
import example, { ThemeContext, ThemeSwitcher } from './example'

describe('ThemeSwitcher', () => {
  let component: RenderResult

  afterEach(() => {
    cleanup()
  })

  describe('example', () => {
    beforeEach(() => {
      component = render(example)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('when switching themes', () => {
    let select: HTMLElement
    let debug: HTMLElement

    const DebugTheme: FC = () => {
      const theme = useContext(ThemeContext)
      return <pre>current theme: {theme}</pre>
    }

    beforeEach(() => {
      component = render(
        <ThemeSwitcher>
          <DebugTheme />
        </ThemeSwitcher>,
      )
      select = component.getByRole('combobox')
      debug = component.getByText(/current theme:/)
    })

    it('should show default', () => {
      expect(debug).toHaveTextContent('current theme: default')
    })

    describe('when using red', () => {
      beforeEach(() => {
        fireEvent.change(select, { target: { value: 'red' } })
      })

      it('should show red', () => {
        expect(debug).toHaveTextContent('current theme: red')
      })

      describe('when using green', () => {
        beforeEach(() => {
          fireEvent.change(select, { target: { value: 'green' } })
        })

        it('should show green', () => {
          expect(debug).toHaveTextContent('current theme: green')
        })
      })
    })
  })
})
