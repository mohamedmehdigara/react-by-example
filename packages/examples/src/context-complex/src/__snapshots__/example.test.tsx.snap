// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher example should snapshot 1`] = `
<div>
  <label>
    theme
    <select>
      <option>
        default
      </option>
      <option>
        red
      </option>
      <option>
        green
      </option>
    </select>
  </label>
  <br />
  <button
    class="button"
    type="button"
  >
    themed button
  </button>
</div>
`;
